README.md

```diff
@@ -53,7 +56,7 @@ In one sentence, explain why.
 
 ## A1.2
 
-«replace this with your answer»
+O(n). Because it should check if the configuration is an odd-length palindrome; one way to do it is using one pointer goes from left to right and one from right to left, they should meet at the center so we need n / 2 comparisions, which is O(n). 
```

Comment:
# -2

The answer is O(n), but you've considered runtime speed, not memory use. In fact, the runtime speed for a parser is likely to be between O(n<sup>2</sup>) and O(n<sup>3</sup>).

The parser takes its input one token at a time. It doesn't know where the end is
until it gets there. So the mechanical parsing process is to get the next token,
push it on a stack, and then compare the rest of the input. As a result, the
stack can grow to at least n/2, so the memory consumption will be O(n)